### PR TITLE
feat(tabs): add option to disable pagination

### PR DESCRIPTION
**Pull request by tiaguinho**
issue https://github.com/angular/material2/issues/9162

### CODE REVIEW

**Comment by tiaguinho** [comment]
@andrewseguin thoughts?

File: src/lib/tabs/tab-header.ts [hunk 1]
```diff
@@ -379,7 +382,7 @@ export class MatTabHeader extends _MatTabHeaderMixinBase
    * should be called sparingly.
    */
   _checkPaginationEnabled() {
-    const isEnabled =
+    const isEnabled = this.disablePagination ? !this.disablePagination :
```
Comment (andrewseguin) [inline src/lib/tabs/tab-header.ts]: Replace `!this.disablePagination` with just `false`

**Comment by andrewseguin** [comment]
Can you add some testing to the spec and include it in the demo app to validate its usage/ergonomics? Would be nice to see it in action and make sure it makes sense to use.

**Comment by tiaguinho** [comment]
@andrewseguin just add test and include in the demo app. If need more changes, please let me know.

File: src/lib/tabs/tab-group.ts [hunk 2]
```diff
@@ -119,6 +119,9 @@ export class MatTabGroup extends _MatTabGroupMixinBase implements AfterContentIn
   /** Position of the tab header. */
   @Input() headerPosition: MatTabHeaderPosition = 'above';
 
+  /** Disable tab pagination */
+  @Input() disablePagination: boolean;
```
Comment (willshowell) [inline src/lib/tabs/tab-group.ts]: Consider coercing this to a boolean (see `dynamicHeight`) so users could do 

```html
<mat-tab-group disablePagination>
  <!-- ... -->
</mat-tab-group>
```

**Comment by tiaguinho** [comment]
@willshowell I just change the code as you said.

**Comment by tiaguinho** [comment]
@andrewseguin Need more changes?

File: src/lib/tabs/tab-group.ts [hunk 3]
```diff
@@ -119,6 +119,12 @@ export class MatTabGroup extends _MatTabGroupMixinBase implements AfterContentIn
   /** Position of the tab header. */
   @Input() headerPosition: MatTabHeaderPosition = 'above';
 
+  /** Disable tab pagination */
+  @Input()
+  get disablePagination(): boolean { return this._disablePagination; }
+  set disablePaignation(value: boolean) { this._disablePagination = coerceBooleanProperty(value); }
```
Comment (andrewseguin) [inline src/lib/tabs/tab-group.ts]: Got a bit of a breaking typo here

**Comment by andrewseguin** [comment]
While running the demo app, I see that if `Disable Pagination` is set and the tabs overflow, there is some jank when selecting the last tab shown. It looks like the header tries to animate over to show the selected tab but then animates again to scroll left.

Are you seeing the same behavior?

**Comment by jelbourn** [comment]
Closing due to inactivity

**Pull request closed without merging**